### PR TITLE
Fix bug462 telemetry naming miss instrumentation key

### LINKED ISSUE #462: Telemetry Naming not align with .net Version
# Issue Description

when I use the Java sdk, the log sent out is named as
 
**"name": "Microsoft.ApplicationInsights.Request"**

but when I use .net sdk, the name contains the instrumentation key, like:

**"name": "Microsoft.ApplicationInsights.aif4614a93a841f4d9dae2754a1d946fc8b.Message",**

# Impact

While using the ApplicaitonInsights Java SDK as FirstParty Partner to onboard Microsoft Asimove Platform, no source node will be created, as the Event Name does not contains the Key

Ref: 
[Xpert Ticket 2777] (http://asimovtickets/helpdesk/Ticket/2777)

# Root Cause
The root cause is that Java SDK hard corded the Telemetry Name, but .net SDK process it with give prefix:

```
        /// <summary>
        /// Normalize instrumentation key by removing dashes ('-') and making string in the lowercase.
        /// In case no InstrumentationKey is available just return empty string.
        /// In case when InstrumentationKey is available return normalized key + dot ('.')
        /// as a separator between instrumentation key part and telemetry name part.
        /// </summary>
        private static string NormalizeInstrumentationKey(string instrumentationKey)
        {
            if (instrumentationKey.IsNullOrWhiteSpace())
            {
                return string.Empty;
            }

            return instrumentationKey.Replace("-", string.Empty).ToLowerInvariant() + ".";
        }
```

### PR DESCRIPTION
**Pull request by XiaoliangHuang**
Fix #462 telemetry name miss instrumentation key.

For significant contributions please make sure you have completed the following items:

- [x] Design discussion issue #
- [x] Changes in public surface reviewed
- [ ] CHANGELOG.md updated

### CODE REVIEW

File: core/src/test/java/com/microsoft/applicationinsights/telemetry/BaseTelemetryTest.java [hunk 1]
```diff
@@ -23,11 +23,17 @@
...
 import com.microsoft.applicationinsights.internal.schemav2.Data;
 import com.microsoft.applicationinsights.internal.schemav2.Domain;
+import com.microsoft.applicationinsights.internal.schemav2.Envelope;
+import com.microsoft.applicationinsights.internal.schemav2.SessionStateData;
+
 import org.junit.Test;
 
 import java.io.IOException;
+import java.io.StringWriter;
+import java.lang.reflect.*;
```
Comment (dhaval24) [inline core/src/test/java/com/microsoft/applicationinsights/telemetry/BaseTelemetryTest.java]: @XiaoliangHuang can we avoid use of wild cards for import here?

File: core/src/main/java/com/microsoft/applicationinsights/telemetry/BaseTelemetry.java [hunk 2]
```diff
@@ -184,4 +190,23 @@ protected String getEnvelopName() {
     protected String getBaseTypeName() {
         throw new UnsupportedOperationException();
     }
+    
+    private String normalizeInstrumentationKey(String instrumentationKey){
+    	if (StringUtils.isNullOrEmpty(instrumentationKey)){
+    		return "";
+    	}
+    	else{
+    		return instrumentationKey.replace("-", "").toLowerCase() + ".";
```
Comment (dhaval24) [inline core/src/main/java/com/microsoft/applicationinsights/telemetry/BaseTelemetry.java]: @XiaoliangHuang if in rare case instrumentation key is only "----" (dashes) then this part will introduce additional dot. I don't think this is desirable. Please add a check and a test case also.

File: core/src/main/java/com/microsoft/applicationinsights/telemetry/BaseTelemetry.java [hunk 3]
```diff
@@ -21,6 +21,7 @@
 
 package com.microsoft.applicationinsights.telemetry;
 
+import com.microsoft.applicationinsights.agent.internal.common.StringUtils;
```
Comment (dhaval24) [inline core/src/main/java/com/microsoft/applicationinsights/telemetry/BaseTelemetry.java]: @XiaoliangHuang can we avoid agent import in core classes. I believe there is similar method in core also.

Comment (XiaoliangHuang) [reply core/src/main/java/com/microsoft/applicationinsights/telemetry/BaseTelemetry.java]: updated, the agent dependent on the core, should not refer agent from core.

File: core/src/main/java/com/microsoft/applicationinsights/telemetry/BaseTelemetry.java (hunk 2 above)
Comment (XiaoliangHuang) [reply core/src/main/java/com/microsoft/applicationinsights/telemetry/BaseTelemetry.java]: Add special case handler, if there are only characters from "-/.", then do not contains ikey.
maybe in the future, should add some schema check on the configuration to restrict the usage of special characters in the ikey

File: core/src/test/java/com/microsoft/applicationinsights/telemetry/BaseTelemetryTest.java (hunk 1 above)
Comment (XiaoliangHuang) [reply core/src/test/java/com/microsoft/applicationinsights/telemetry/BaseTelemetryTest.java]: this was added as I planed to use reflect to test the private method, but at last not used, so I removed it.

File: core/src/main/java/com/microsoft/applicationinsights/telemetry/BaseTelemetry.java [hunk 4]
```diff
@@ -184,4 +191,23 @@ protected String getEnvelopName() {
     protected String getBaseTypeName() {
         throw new UnsupportedOperationException();
     }
+    
+    private String normalizeInstrumentationKey(String instrumentationKey){
+    	if (StringUtils.isEmpty(instrumentationKey) || StringUtils.containsOnly(instrumentationKey, ".- ")){
```
Comment (littleaj) [inline core/src/main/java/com/microsoft/applicationinsights/telemetry/BaseTelemetry.java]: Is it possible for it to have a missing iKey at this point? or is this an error scenario?

Comment (SergeyKanzhelev) [reply core/src/main/java/com/microsoft/applicationinsights/telemetry/BaseTelemetry.java]: outcome will be questionable if key is empty, but it is definitely possible. So thins handling is OK

Comment (littleaj) [reply core/src/main/java/com/microsoft/applicationinsights/telemetry/BaseTelemetry.java]: cool. thanks for clarifying

**Comment by dhaval24** [comment]
@grlima this should be good for merging. Please have a look at it once before we do so.

**Comment by dhaval24** [comment]
@XiaoliangHuang Thank you very much for your contribution towards the SDK. Your pull request is now merged.

**Pull request merged**